uPortal-api/uPortal-api-rest/src/main/java/org/apereo/portal/rest/search/PortletsSearchStrategy.java

```diff
@@ -86,12 +86,12 @@ public String getResultTypeName() {
 
     @Override
     public List<?> search(String query, HttpServletRequest request) {
-
+        logger.debug("Entering search() with query={}", query);
         final List<Object> rslt = new ArrayList<>();
 
         try (IndexReader indexReader = DirectoryReader.open(directory)) {
-
-            final Query q = queryParser.parse(query);
+            final String queryString = query.endsWith(" ") ? query : query + "*";
```

Comment:
If I'm reading this correctly it will add a wildcard if a user is typing and has reached the end of a word for example "notification ", but would not add a wildcard for midword typing such as "notificatio"?
Is that expected? Would we want a wildcard in that case?

Reply:
why not adding a wildcard all the time ? 
I think that users won't understand (me first) that they should add a space at the end of a word to have a largest search. Also how it will works if the user search on "notifica*" already with a wildcard ?

Reply:
I will have to test more, but I believe double asterisks will not be an issue. We don't want an asterisk alone because returning everything indexed is not a very good search result.

Reply:
> I will have to test more, but I believe double asterisks will not be an issue.

Are you saying there is already an asterisk on text that doesn't end with a space?
If so, would it make sense to update that code to always add an asterisk?

> We don't want an asterisk alone because returning everything indexed is not a very good search result.

Right, if we're looking to catch an empty string, would a condition like `query.trim().length > 0` make more sense?

Reply:
What the code is doing is adding an asterisk mid-word.

Example 1: "cache " -> ends with space -> "cache " -> 'cache' as token
Example 2: "notif" -> does not end with space -> "notif*"  -> 'notif*' is the token

I wan to avoid "cache " -> "cache *" -> 'cache' and '*' tokens

Searches of empty string are not allowed / submitted. Lucene is pretty good about the basics. It just leaves wildcards as an explicit. My research to enable partial search tokens without tokens has suggested simply adding a trailing asterisk.

Reply:
Ah, that makes more sense, thanks for the clarification

Reply:
Ok, It's good for me too !
But I'm guessing how users will understand that we need to add a space at the end of the word to make search on the exact word. Is there something to help to understand that ? like a placeholder ? a tooltip ? any other thing to help ?